internal/graph/graph.go

```diff
@@ -332,6 +335,61 @@ nextSample:
 	return selectNodesForGraph(nodes, o.DropNegative)
 }
 
+// Trims a Graph that is in forest form to contain only the nodes in kept. This
+// will not work correctly in the case that a node has multiple parents.
+func (g *Graph) TrimTree(kept NodeSet) *Graph {
```

Comment:
Can you receive a map[*Node]bool as a parameter instead of a NodeSet? That way we wouldn't need to change NodeSet.